basesandbox/src/ipc/mod.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pub mod domain_socket;
+pub mod intra;
+pub mod semaphore;
+use std::collections::hash_map::HashMap;
+
+/// Inter-process obejct that is constructed / synchronized by another Inter-process mechanism.
```

Comment:
The overall naming and terms used in the comments disturb me. Inter-process itself means nothing about communication. So if you mean IPC, then use IPC. Not just `inter-process`.

In addition to the above point, there is a typo. ;) **obejct** --> object.

Reply:
`InterProcessUnit` itself has nothing to do with IP'C'. It's just inter-processely initializable object. IPC endpoint is one possible implementor of `InterProcessUnit`